File: 1790_check_if_one_string_swap_can_make_strings_equal.cpp
class Solution {
public:
    bool areAlmostEqual(string a, string b) {
        if (a == b) return true;
        int n = a.size();
        string c = b;
        for (int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {
                swap(c[i], c[j]);
                if (c == a) return true;
                c = b;
            }
        }
        return false;
    }
};

File: 1790_check_if_one_string_swap_can_make_strings_equal.py
class Solution(object):
    def areAlmostEqual(self, s1, s2):
        if s1 == s2:
            return True
        n = len(s1)
        s2_list = list(s2)
        for i in range(n - 1):
            for j in range(i + 1, n):
                s2_list[i], s2_list[j] = s2_list[j], s2_list[i]
                if ''.join(s2_list) == s1:
                    return True
                s2_list = list(s2) 
        return False
